packages/web3-eth/test/integration/subscription_logs.test.ts

```diff
@@ -51,43 +51,42 @@ const makeFewTxToContract = async ({
 };
 describeIf(isWs)('subscription', () => {
 	let clientUrl: string;
-	let accounts: string[] = [];
 	let web3Eth: Web3Eth;
 	let providerWs: WebSocketProvider;
+	let contractInstance: Contract<typeof BasicAbi>;
 	let contract: Contract<typeof BasicAbi>;
 	let deployOptions: Record<string, unknown>;
 	let sendOptions: Record<string, unknown>;
 	let from: string;
 	const testDataString = 'someTestString';
-	beforeAll(async () => {
+	let tempAcc: { address: string; privateKey: string };
+
+	beforeEach(async () => {
+		tempAcc = await createTempAccount();
+	});
+	beforeAll(() => {
 		clientUrl = getSystemTestProvider();
-		accounts = await getSystemTestAccounts();
-		[, from] = accounts;
-		providerWs = new WebSocketProvider(
-			clientUrl,
-			{},
-			{ delay: 1, autoReconnect: false, maxAttempts: 1 },
-		);
-		contract = new Contract(BasicAbi, undefined, {
+		providerWs = new WebSocketProvider(clientUrl);
+		contractInstance = new Contract(BasicAbi, undefined, {
 			provider: clientUrl,
 		});
-
-		deployOptions = {
-			data: BasicBytecode,
-			arguments: [10, 'string init value'],
-		};
-
-		sendOptions = { from, gas: '1000000' };
-
-		contract = await contract.deploy(deployOptions).send(sendOptions);
 	});
 	afterAll(() => {
 		providerWs.disconnect();
+		(contractInstance.provider as WebSocketProvider).disconnect();
 	});
 
 	describe('logs', () => {
 		it(`wait for ${checkEventCount} logs`, async () => {
 			web3Eth = new Web3Eth(providerWs as Web3BaseProvider);
+			from = tempAcc.address;
```

Comment:
Declaring this here instead of above. Because it is assigned here.
```suggestion
			const from = tempAcc.address;
```

Reply:
done